src/core/net/dns_headers.cpp

```diff
@@ -149,6 +147,7 @@ otError Name::AppendMultipleLabels(const char *aLabels, Message &aMessage)
                 ExitNow();
             }
 
+            VerifyOrExit(index + 1 < kMaxEncodedLength, error = OT_ERROR_INVALID_ARGS);
```

Comment:
Explain: `index + 1` should always equals to the actual encoded name length after the next statement (i.e. AppendLabel) is executed. So this condition checks the exact condition we are looking for: `encoded length < 255`.